chia/data_layer/download_data.py

```diff
@@ -138,6 +166,52 @@ async def write_files_for_root(
     return WriteFilesResult(written, filename_full_tree if written_full_file else None, filename_diff_tree)
 
 
+async def download_file(
+    target_filename_path: Path,
+    store_id: bytes32,
+    root_hash: bytes32,
+    generation: int,
+    server_info: ServerInfo,
+    proxy_url: str,
+    downloader: Optional[PluginRemote],
+    timeout: int,
+    client_foldername: Path,
+    log: logging.Logger,
+    grouped_by_store: bool,
+    group_downloaded_files_by_store: bool,
+) -> bool:
+    if target_filename_path.exists():
+        return True
+    filename = get_delta_filename(store_id, root_hash, generation)
+    if grouped_by_store:
+        filename = filename.replace("-", "/", 1)
```

Comment:
this is the name we request to the servers, as opposed to what we store locally to need a Path. perhaps extend `get_delta_filename` to consider `group_by_store` would be more appropiate?

Reply:
Sounds reasonable.  Or avoid the difference and when you want the name use `.name`?  Either way

Reply:
Let me try to say that a bit more clearly.  This PR adds the `get_(full_tree|delta)_filename_path()` functions.  Perhaps those should just replace the existing functions and uses of the existing functions would just use `.name` on the result?  This would avoid the matrixing of the functions against filename vs. path returns.

Reply:
We discussed this and Florin pointed out various complexities I was overlooking.  Some improvements were made in centralizing the format definitions.